test/models/thesis_test.rb

```diff
@@ -1318,17 +1318,33 @@ def attach_file_with_purpose_to(thesis, purpose = 'thesis_pdf')
     thesis = theses(:one)
     assert_not_includes Thesis.not_consented_to_proquest, thesis
 
+    # multi-author thesis that has opted in is excluded
+    thesis = theses(:two)
+    assert_not_includes Thesis.not_consented_to_proquest, thesis
+
     # multi-author thesis with all opt-outs is included
     thesis = theses(:with_hold)
     assert_includes Thesis.not_consented_to_proquest, thesis
 
-    # multi-author thesis with one opt-in and one opt-out is excluded
+    # multi-author thesis with one opt-in and one opt-out is included
     thesis = theses(:doctor)
-    assert_not_includes Thesis.not_consented_to_proquest, thesis
+    assert_includes Thesis.not_consented_to_proquest, thesis
 
-    # multi-author thesis that has opted in is excluded
-    thesis = theses(:two)
-    assert_not_includes Thesis.not_consented_to_proquest, thesis
+    # multi-author thesis with one opt-in and one null is included
+    assert_equal 2, thesis.authors.count
```

Comment:
It feels like we are inheriting the thesis from the pervious set of assertions and not loading one intentionally. The test is likely valid as-is, but if someone inserted a different set of assertions between the previous and this one that loaded or changed state of a thesis/authors we'd be confused here. I'd suggest we load a fixture for a thesis for each case we are testing even if we end up reloading the same fixture multiple times. The prevent this potential leakage, it may (or may not!) be useful to use more explicit variable names than `thesis`.  It's really easy to let data slip between assertions when we reuse the variable name (which I think we do quite frequently and is not limited to this work but I'm only noticing it now as a potential for introducing bugs).

Reply:
@JPrevost I totally hear you. In this case, the impetus was expediency. If I were to add a couple fixtures (or even change existing ones), I would have to update several tests in the report model that expect specific numbers of entries in a table. I fixed a few of those tests the last time around out of frustration, but there are still a handful of them, and my goal was to land this as quickly as possible.

Your point is well taken, though, and I opened this ticket to fix it up: https://mitlibraries.atlassian.net/browse/ETD-621

I also opened this ticket to refactor the bad tests in the report model: https://mitlibraries.atlassian.net/browse/ETD-620